frontend/model/contracts/group.js

```diff
@@ -697,7 +768,29 @@ DefineContract({
         // }
       }
     },
-
+    'gi.contracts/group/resetMonth': {
+      validate: optional(string),
+      process ({ meta }, { state, getters }) {
+        // Loop through missing monthly cycle events that happen before the 'event' parameter's cycle
+        let lastEvent = state.distributionEvents[state.distributionEvents.length - 1]
+        // Fills in multiple missing months when `while` instead of `if`.
+        while (compareCycles(meta.createdDate, lastEvent.data.when) > 0) {
+          // Add the missing monthly cycle event
+          const monthlyCycleEvent = {
+            type: 'startCycleEvent',
+            data: {
+              latePayments: [], // List to be populated later, by the events-parser
+              when: dateToMonthstamp(addMonthsToDate(dateToMonthstamp(lastEvent.data.when), 1))
+            }
```

Comment:
AFAICT, if on line 783 we use a `when` = `meta.createdDate` instead of using `addMonthsToDate`, then this `while` loop can be replaced with an `if`.

Is there any reason to not do that?

Reply:
```// Fills in multiple missing months when `while` instead of `if`.```

Think of the ``distributionEvents`` as a way of saving the previous two month's, in addition to the current month's information (such as ```latePayments```). If we don't have this information being propagated and preserved here, it will have to be inferred (or worse: entirely lost) later when the final distribution function is called.

I have changed many cases of using an ```while``` to being ```if```, for now... I 100% believe this results in the problem mentioned above, and edge-case problems mentioned on our calls (even though they have a rare probability of occurring)!

I think we might need to keep this comment open issue so we don't forget about it and we can discuss it further, while we try to create cypress/mocha tests that can actually prove these rare edge-cases exist.

Reply:
OK, trust you on this.